### ISSUE TITLE
FE scale audit: Notebooks share menu org list is not paginated

### ISSUE THREAD

**Issue by philipp-spiess** [sourcegraph/sourcegraph#43857]
It's possible to share notebooks with orgs. This shows a dropdown UI of orgs that is currently not paginated and loads all orgs at once. It's possible for large customers to use orgs as a team-level access control feature so we can expect a user to belong to many of them.

https://www.figma.com/file/SiFt7KlbjUwSMXPfsXXdew/Strategic-readiness---design?node-id=45%3A16

**Comment by philipp-spiess** [comment]
Linking some context regarding how to proceed with this issues from Slack: https://sourcegraph.slack.com/archives/C03MW9T8G6A/p1668531004016079

**Comment by novoselrok** [comment]
I don't think we have a ready-made dropdown component that already supports pagination or filtering. So that would require a pretty big design change. Also, afaik, customers do not utilize orgs all that much. It's rare to have more than 1 or 2. With that in mind, I don't think this is a huge problem at the moment.
